### DISCUSSION TITLE
partitioned analysis with mixturefinder?

### DISCUSSION THREAD

**Discussion by DerekSikes** [iqtree/iqtree2#339] [Q&A] [answered]
I want to do a partitioned analysis - DNA + morphology. I was able to follow the instructions to do this but those instructions have the models specified in the nexus partition file.

I would rather have iqtree run mixturefinder on the DNA partition and do the normal morphology model testing on the other. Is this possible?

Also, I already ran mixturefinder on the DNA partition in isolation. Is there a way to feed these results into the partitioned DNA+morphology analysis (eg to specify the DNA model as

MIX{GTR+FO,TVM+FO,TPM2u+FO,TIM2+FO,HKY+FO}+I+R4

**Comment by DerekSikes** [comment]
Update, I see that in the partitions file "The line "charpartition" is not mandatory. In the absence of this command, IQ-TREE will use all "charset" as partitions and find the best-fit model for each one of them."

This invokes modelfinder for each partition (DNA vs morph) but my DNA is a concatenation of 4 genes, and 3 are protein coding. When I ran mixturefinder on these it found 5 classes of data so I want to use that for analysis of the DNA. Is that possible in a partitioned analysis. Can I specify 

-m MIX+MFP -B 1000 -alrt 1000

somehow? Maybe in the partition file?

**Reply by HuaiyanRen** [reply]
Currently, MixtureFinder can't assign a site in the alignments to a specific class of mixture. Because the site-likelihood is calculated under all classes.

There are an option in IQ-TREE 2 may relevant to this issue.

> -wspm: Write site posterior probabilities per mixture class to .siteprob file.

Say if we have a three-class mixture model, then we look at the .siteprob file, the first three rows may be:
```
Site    p1    p2    p3
1     0.998    0.001    0.001
2     0.001    0.998    0.001
```
Then, it seems like site 1 is in partition 1 and site 2 is in partition 2. But, there will be lot of sites like this:
```
Site    p1    p2    p3
m     0.5    0.1    0.4
n     0.333    0.333    0.334
```
We can't directly put all these sites in some specific "partitions".

Hidden Markov Model (HMM) could be a solution of this problem. But in IQ-TREE it only implemented in tree-mixture model. 

So, if you really need to partition the alignment according to the mixture model, you can print out .siteprob and try other methods.

**Reply by roblanf** [reply]
What I understand from this @HuaiyanRen is that the answer to the original question is: No, you cannot yet combine DNA and morphology, using a mixture model for the DNA.

Is that correct?

**Reply by HuaiyanRen** [reply, marked as answer]
Yes, it's correct. MixtureFinder is only for DNA, currently not available for amino acid or morphology.

**Reply by DerekSikes** [reply]
Thanks for the answers. I will do a mixturefinder search with the DNA alone and then take that tree as a constraint tree to run the morphology (since there are many species I have morphology for but no DNA, this will place those species on the DNA constraint tree). I will compare these results to a combined DNA+morphology partitioned (non mixturefinder) analysis.

**Comment by roblanf** [comment]
@HuaiyanRen this is one for you!
